package.json

```diff
@@ -40,28 +40,25 @@
     },
     "dependencies": {
         "@azure/functions": "^1.2.3",
-        "@types/lodash": "^4.14.119",
-        "@types/node": "10.14.2",

```

Comment:
This one is the most complicated. There is one reference to a Node.js type (NodeJS.Timer), but I still don't think that means we should list `@types/node` as a dependency. A few reasons:

1. The durable-functions package appears to support Node 10,12,14, & 16 so forcing users to use v10 would be very limiting and often incorrect. Users need to specify the version of node types they are actually running on in their own package.json, which will likely not be 10 because it's pretty old.
1. Most users will include `@types/node` in their own package.json, so it's not like the type for `NodeJS.Timer` will be missing
2. Long term I would recommend you try to remove this reference to improve you compatibility across Node versions, but realistically I doubt the `NodeJS.Timer` type has changed in forever

<img width="1586" alt="Screen Shot 2022-04-22 at 2 44 01 PM" src="https://user-images.githubusercontent.com/11282622/164802491-8c7daadd-69fd-4bef-82a2-bb0c2998564e.png">

Reply:
Don't we need some reference to `@types/node` to annotate the sleep method? Not saying that it's worth it for a single method, just trying to understand how this works is working.

Reply:
It depends what you mean by "need" or "work". Once the typescript files get compiled to javascript, the types don't matter at all so the underlying javascript will always work. The only thing they matter for is a user's build step and that will fail if users don't have `@types/node` in their package.json.

However, the build will pass if a user adds `@types/node` in their _own_ package.json, and my point was the vast majority of users have already done that. It would actually be very difficult to create a Node.js app that _didn't_ reference node types haha (as soon as you reference `path`, `http`, `fs`, `process`, `os` etc. you'll need the types). Even in the rare case they don't have the types, I think the error is pretty self explanatory (or at least, TypeScript users will hit this many times in their life and know how to fix):

```
node_modules/durable-functions/lib/src/utils.d.ts:1:23 - error TS2688: Cannot find type definition file for 'node'.
 
1 /// <reference types="node" />
```

Reply:
I see. I do know that the types are erased at compile-time, but for our TS users I do feel a little strange not including a types dependency (like @types/node) whose exclusion will cause a compile type error.

But, I trust you when you say that TS users will know how to deal with this, as well as the fact that most users will get this dependency as soon as they import pretty much anything else. So I'll defer to your expertise with this one and be ok with removing it :)